pep-0221.txt

```diff
@@ -5,113 +5,120 @@ Last-Modified: $Date$
-    in that it cannot accommodate an `as' clause, and that no STORE
-    opcodes are generated; the objects imported are loaded directly
-    into the local namespace. This also means that names imported in
-    this fashion are always local, and do not follow the `global'
-    directive.
-    
-    An additional change to this syntax has also been suggested, to
-    generalize the expression given after the `as' clause.  Rather
-    than a single name, it could be allowed to be any expression that
-    yields a valid l-value; anything that can be assigned to.  The
-    change to accommodate this is minimal, as the patch[2] proves, and
-    the resulting generalization allows a number of new constructs
-    that run completely parallel with other Python assignment
-    constructs. However, this idea has been rejected by Guido, as
-    `hypergeneralization'.
+stores the module ``os`` locally as ``os``, so that the imported
+submodule ``path`` is accessible as ``os.path``.  As a result::
+
+    import os.path as p
+
+stores ``os.path``, not ``os``, in ``p``.  This makes it effectively the
+same as::
+
+    from os import path as p
+
+
+Implementation details
+======================
+
+This PEP has been accepted, and the suggested code change has been
+checked in.  The patch can still be found in the SourceForge patch
+manager [1]_.  Currently, a ``NAME`` field is used in the grammar rather
+than a bare string, to avoid the keyword issue.  It introduces a
+new bytecode, IMPORT_STAR, which performs the ``from module import
+\*`` behaviour, and changes the behaviour of the ``IMPORT_FROM``
+bytecode so that it loads the requested name (which is always a
+single name) onto the stack, to be subsequently stored by a STORE
+opcode. As a result, all names explicitly imported now follow the
+``global`` directives.
+
+The special case of ``from module import *`` remains a special case,
+in that it cannot accommodate an ``as`` clause, and that no STORE
+opcodes are generated; the objects imported are loaded directly
+into the local namespace. This also means that names imported in
+this fashion are always local, and do not follow the ``global``
+directive.
+
+An additional change to this syntax has also been suggested, to
+generalize the expression given after the ``as`` clause.  Rather
+than a single name, it could be allowed to be any expression that
+yields a valid l-value; anything that can be assigned to.  The
+change to accommodate this is minimal, as the patch [2]_ proves, and
+the resulting generalization allows a number of new constructs
+that run completely parallel with other Python assignment
+constructs. However, this idea has been rejected by Guido, as
+"hypergeneralization".
 
 
 Copyright
+=========
 
-    This document has been placed in the Public Domain.
+This document has been placed in the Public Domain.
 
 
 References
+==========
+
+.. [1] http://sourceforge.net/patch/?func=detailpatch&patch_id=101135&group_id=5470
```

Comment:
For [1], the actual commit is https://github.com/python/cpython/commit/e6ba0ed982fd756f94833696e493544e4d20ac8b (we probably need to use the changeset from hg.python.org since the GitHub link will be broken when we switched to GitHub)